src/builtins.ts

```diff
@@ -665,29 +665,20 @@ export function compileCall(
       let type = evaluateConstantType(compiler, typeArguments, operands, reportNode);
       compiler.currentType = Type.bool;
       if (!type) return module.unreachable();
-      return module.i32(type.kind === TypeKind.VOID ? 1 : 0);
+      return module.i32(type.kind == TypeKind.VOID ? 1 : 0);
     }
     case BuiltinSymbols.lengthof: { // lengthof<T>(): i32
       let type = evaluateConstantType(compiler, typeArguments, operands, reportNode);
       compiler.currentType = Type.i32;
-      if (type === null) {
-        compiler.error(
-          DiagnosticCode.Expected_0_type_arguments_but_got_1,
-          reportNode.range, "1", (typeArguments ? typeArguments.length : 1).toString(10)
-        );
-        return module.unreachable();
-      }
+      if (type === null) return module.unreachable();
 
-      if (type.signatureReference === null) {
-        compiler.error(
-          DiagnosticCode.Type_0_has_no_call_signatures,
-          reportNode.range, "1", (typeArguments ? typeArguments.length : 1).toString(10)
-        );
-        return module.unreachable();
-      }
-      return type.signatureReference.parameterNames === null
+      let signatureReference = type.signatureReference;
+      if (signatureReference === null) return module.unreachable();
```

Comment:
This does not report, so the previous error was correct.

Reply:
Oh. Whoops.

Reply:
I reverted this. All set.